src/AppInstallerRepositoryCore/Microsoft/SQLiteIndexSource.cpp

```diff
@@ -68,26 +87,70 @@ namespace AppInstaller::Repository::Microsoft
                 }
             }
 
-            std::vector<Utility::VersionAndChannel> GetVersions() override
+            SQLiteIndex::IdType m_manifestId;
+        };
+
+        // The IPackage impl for SQLiteIndexSource.
+        struct Package : public SourceReference, public IPackage
+        {
+            Package(const std::shared_ptr<const SQLiteIndexSource>& source, SQLiteIndex::IdType idId) :
+                SourceReference(source), m_idId(idId) {}
+
+            // Inherited via IPackage
+            std::shared_ptr<IPackageVersion> GetInstalledVersion() const override

```

Comment:
>GetInstalledVersion() [](start = 45, length = 21)

Question outside of this PR, will there be GetInstalledInstallerType() or something similar under this interface? Or it'll be from other interfaces?

Reply:
Yes, my intention is to record such data when we install, or to reproduce it when we read from the system.  The list of properties will be dynamic, so adding new ones should be easy as well.

---
In reply to: [492441969](https://github.com/microsoft/winget-cli/pull/575#discussion_r492441969) [](ancestors = 492441969)